xdsl/backend/riscv/lowering/convert_memref_to_riscv.py

```diff
@@ -264,10 +282,12 @@ class ConvertMemrefToRiscvPass(ModulePass):
     name = "convert-memref-to-riscv"
 
     def apply(self, ctx: MLContext, op: ModuleOp) -> None:
+        contains_malloc = PatternRewriteWalker(ConvertMemrefAllocOp()).rewrite_module(
+            op
+        )
```

Comment:
Nit: I think I would prefer to track "inserted malloc op" as an attribute on the rewrite for slightly cleaner code. Especially since we will need to repeat this for free and realloc.

Reply:
Agree to disagree :)